thriftbp/headers.go

```diff
@@ -6,40 +6,41 @@ import (
 	"github.com/apache/thrift/lib/go/thrift"
 
 	"github.com/reddit/baseplate.go/ecinterface"
+	"github.com/reddit/baseplate.go/transport"
 )
 
 // Edge request context propagation related headers, as defined in
 // https://pages.github.snooguts.net/reddit/baseplate.spec/component-apis/thrift#edge-request-context-propagation
 const (
-	HeaderEdgeRequest = "Edge-Request"
+	HeaderEdgeRequest = transport.HeaderEdgeRequest
```

Comment:
please also mark them as deprecated if possible, that way staticcheck will complain if people tries to use them.

I'm not super sure if `// Deprecated: ...` on the whole block works or you have to put it on every constant. but you can verify that by trying to use them and see if staticcheck complains.

Reply:
Do we have an example where we used this in the past, I couldn't find anything searching through `git log -p`. I tried several versions but staticcheck would not complain that it is still being used within `thriftbp` or does it only apply if you import a deprecated type?

i.e. with
```go
// Deprecated: HeaderUserAgent related headers.
const HeaderUserAgent = "User-Agent"
```
it would result in
```
thriftbp/headers.go:46:1: comment on exported const HeaderUserAgent should be of the form "HeaderUserAgent ..." (ST1022)
```

Reply:
you can have both:
```go
// HeaderUserAgent is the user-agent header.
//
// Deprecated: use transport.HeaderUserAgent instead
const HeaderUserAgent = "User-Agent"
```

or if that doesn't work, you can always add linter directive (see https://staticcheck.io/docs/configuration#maintenance-of-linter-directives):
```go
// Deprecated: HeaderUserAgent related headers.
//lint:ignore ST1022 this is deprecated
const HeaderUserAgent = "User-Agent"
```